jerry-core/ecma/operations/ecma-regexp-object.h

```diff
@@ -18,6 +18,12 @@
 
 #ifndef CONFIG_DISABLE_REGEXP_BUILTIN
 
+#ifdef REGEXP_RECURSION_LIMIT
+#define INCREASE_RECURSION_DEPTH_COUNTER (++re_ctx_p->recursion_depth)
+#else
+#define INCREASE_RECURSION_DEPTH_COUNTER
```

Comment:
Since this macro performs an action it is usual to add `()` to look like a function. Also the name should contain REGEXP. For example:

`#define REGEXP_INCREASE_RECURSION_COUNTER() (++re_ctx_p->recursion_depth)`

I think depth is misleading here, since the currently used counter represents the "remaining depth".

Perhaps the variable `recursion_depth` could be renamed to `recursion_counter`.